mllib/src/test/scala/org/apache/spark/ml/util/MLTest.scala

```diff
@@ -108,5 +111,29 @@ trait MLTest extends StreamTest with TempDirectory { self: Suite =>
       otherResultCols: _*)(globalCheckFunction)
     testTransformerOnDF(dataframe, transformer, firstResultCol,
       otherResultCols: _*)(globalCheckFunction)
+    }
+
+  def testTransformerByInterceptingException[A : Encoder](
+    dataframe: DataFrame,
+    transformer: Transformer,
+    expectedMessagePart : String,
+    firstResultCol: String) {
+
+    def hasExpectedMessage(exception: Throwable): Boolean =
```

Comment:
I doubt whether the check here is too strict. It require exactly match message so when some class modify the exception message then many testcase will fail.
Or can we just check the exception type ?

Reply:
It uses contains. I would keep this behaviour as the test is more well spoken this way.
@jkbradley?

Reply:
Since most other tests check parts of the message, I'm OK with this setup.  When we don't think the message will remain stable, we can pass an empty string for expectedMessagePart.

Reply:
Just curious: Did you have to add the getCause case because of streaming throwing wrapped exceptions?

Reply:
Yes, that was the reason.